src/models/Log.ts

```diff
@@ -0,0 +1,52 @@
+import { red, yellow, green, dim, magenta, cyan } from 'colors';
+
+type BasicLogHandler = (message: string) => void;
+type DeviceTrafficHandler = (topic: string, payload: any) => void;
+type Label = string;
+type Message = string;
+type MessageEntry = [Label, Message];
+
+export interface Logger {
+  error: BasicLogHandler;
+  info: BasicLogHandler;
+  success: BasicLogHandler;
+  debug: BasicLogHandler;
+  outgoing: DeviceTrafficHandler;
+  incoming: DeviceTrafficHandler;
+  prettify: (header: string, body: MessageEntry[])  => string,
+}
+
+const prettyMessage = (header: string, body: MessageEntry[]): string => `
+************** ${header} ***********
+${body.map(([key, val]) => `${key}: ${val}`).join('\n')}
+**************${'*'.repeat(header.length + 2)}***********
+`;
+
+export class Log implements Logger {
+  constructor(private readonly verbose: boolean) {}
+  error(message: string) { console.log(red(message)); }
+  info(message: string) { console.log(yellow(message)); }
+  success(message: string) { console.log(green(message)); }
+  debug(message: string) { this.verbose && console.log(dim(message)); }
+  outgoing(topic: string, payload: any) {
+    console.log(cyan(prettyMessage(
+        'MESSAGE SENT', [
+          ['TOPIC', topic],
+          ['MESSAGE', JSON.stringify(payload, null, 2)],
+        ]),
+      ),
+    );
+  }
+  incoming(topic: string, payload: any) {
+    console.log(magenta(prettyMessage(
+        'MESSAGE RECEIVED', [
+          ['TOPIC', topic],
+          ['MESSAGE', JSON.stringify(payload, null, 2)],
```

Comment:
If payload is a `Buffer` you need to call `toString()` on it. Currently the sim outputs a buffer array.

![Screen Shot 2020-12-04 at 1 55 36 PM](https://user-images.githubusercontent.com/7103368/101218765-84433e80-3638-11eb-900f-a72ec987fe87.png)

Reply:
John said it should only send out array tuples and it's somehow on the client that messages are being received in buffer format 🤷 

In any case, the log has been updated to handle buffer payloads.